src/main/perf/LineFileDocs.java

```diff
@@ -407,6 +446,8 @@ static Document cloneDoc(Document doc1) {
         doc2.add(new NumericDocValuesField(f.name(), f.numericValue().longValue()));
       } else if (f instanceof BinaryDocValuesField) {
         doc2.add(new BinaryDocValuesField(f.name(), f.binaryValue()));
+      } else if (f instanceof VectorField) {
+        doc2.add(new VectorField(f.name(), ((VectorField) f).vectorValue()));
```

Comment:
Does this make a deep copy of the `float[]`?  Or it is shared?

Reply:
right we will later replace the value; the only reason to pass it here is to preserve its dimension, which becomes fixed in the VectorField's type.